src/endpoint/s3/s3_rest.js

```diff
@@ -274,6 +274,12 @@ function authenticate_request(req, res) {
 }
 
 async function authorize_request(req) {
+    const err = await req.object_sdk.authorize_request_account(req.params.bucket);
+    if (err) throw new S3Error(S3Error[err]);
```

Comment:
I don't think returning error here makes sense - we almost never do this in logical flows.
I suggest that object_sdk will simply throw the appropriate RPCError with code that maps to the correct S3Error in s3_rest.